camkes/templates/macros.py

```diff
@@ -686,32 +686,105 @@ def virtqueue_get_client_id(composition, end, configuration):
 NO_CHECK_UNUSED.add('parse_dtb_node_interrupts')
 
 
-def parse_dtb_node_interrupts(node, max_num_interrupts):
+def parse_dtb_node_interrupts(node, max_num_interrupts, arch):
```

Comment:
What would it look like if you split this into two sub-functions, one for arm and one for riscv. Given that they have different deserialization formats, would this be simpler?

Reply:
I am not sure if this is really helpful in the current state. My understanding is, that the interrupt description format is architecture agnostic. For RISC-V, there seems no use case for the 3-value format, because the 'type' has nothing specified for RISC-V. The 2-value format could be used though if one wants to describe the trigger. All we have at the moment are the SiFive device trees, that is a too small base to judge on what to do. Our MiG-V device tree will basically follow what SiFive does and keep it to the bare minimum that seL4 needs.
I would suggest to use the current way for the near future, as this provides good flexibility and can be adapted easily to whatever device tree will come.

Reply:
Currently I can't easily tell that this doesn't introduce a regression for Arm just by looking at the changes.

> My understanding is, that the interrupt description format is architecture agnostic.

The description format is defined by the parent interrupt controller.  If the parent is a gicv2 then it's one format but if it's a plic0 then it's a different format.  When parsing the node to calculate which interrupts to create, the parser needs to know how to decode the specific interrupt controller type.  Unfortunately, the current parse_dtb_node_interrupts only supports gicv2/gicv3.  If you are adding support for the plic0 then it can be helpful to split the parsing logic into a plic part and gic part.

Reply:
I left the ARM code in now with a comment that it is used to keep the old behavior.